packages/strapi-plugin-content-manager/admin/src/components/InputUID/index.js

```diff
@@ -0,0 +1,245 @@
+import Input from './InputUID';
+
+// There is no need to create additional files for those little components.
+const Wrapper = styled.div`
+  position: relative;
+  padding-bottom: 23px;
+`;
+const InputContainer = styled.div`
+  position: relative;
+`;
+const ErrorMessage = styled(BaseErrorMessage)`
+  padding-top: 10px;
+`;
+const Name = styled(Label)`
+  display: block;
+  text-transform: capitalize;
+  margin-bottom: 1rem;
+`;
+
+// This component should be in buffetjs. It will be used in the media lib.
+// This component will be the strapi custom dropdown component.
+// TODO : Make this component generic -> InputDropdown.
+// TODO : Use the Compounds components pattern
+// https://blog.bitsrc.io/understanding-compound-components-in-react-23c4b84535b5
+const InputUID = ({
+  error: inputError,
+  required,
+  name,
+  value,
+  contentTypeUID,
+  validations,
+  onChange,
+}) => {
+  const { modifiedData, initialData } = useDataManager();
+  const [isLoading, setIsLoading] = useState(false);
+  const [availability, setAvailability] = useState(null);
+  const [isSuggestionOpen, setIsSuggestionOpen] = useState(true);
+  const [label, setLabel] = useState();
+  const debouncedValue = useDebounce(value, 300);
+  const wrapperRef = useRef(null);
+  const initialValue = initialData[name];
+
+  const checkAvailability = async () => {
+    setIsLoading(true);
+    const requestURL = '/content-manager/explorer/uid/check-availability';
+    try {
+      const data = await request(requestURL, {
+        method: 'POST',
+        body: {
+          contentTypeUID,
+          field: name,
+          value: value || null,
+        },
+      });
+      setAvailability(data);
+
+      if (data.suggestion) {
+        setIsSuggestionOpen(true);
+      }
+      setIsLoading(false);
+    } catch (err) {
+      console.error({ err });
+      setIsLoading(false);
+    }
+  };
+
+  const generate = async () => {
+    setIsLoading(true);
+    const requestURL = '/content-manager/explorer/uid/generate';
+    try {
+      const { data } = await request(requestURL, {
+        method: 'POST',
+        body: {
+          contentTypeUID,
+          field: name,
+          data: modifiedData,
+        },
+      });
+      onChange({ target: { name, value: data, type: 'text' } });
+      setIsLoading(false);
+    } catch (err) {
+      console.error({ err });
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (!value && required) {
+      generate();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (debouncedValue && debouncedValue !== initialValue) {
+      checkAvailability();
+    }
+    if (!debouncedValue) {
+      setAvailability(null);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [debouncedValue]);
+
+  useEffect(() => {
+    let timer;
+
+    if (availability && availability.isAvailable) {
+      timer = setTimeout(() => {
```

Comment:
Why don't you use the `useDebounce` hook here? If you change your hook so it can accept more than one dependency it should work.

Reply:
I don't think that the `useDebounce` hook is defined for this use case. This hook creates another state to return the debounced value (We don't need a new state to disable this label). Maybe we should keep in mind the single responsibility principle for hooks too?

Reply:
Sure